migration-guides/0.14-Migration-Guide.md

```diff
@@ -0,0 +1,596 @@
+
+```diff
+- import Data.Tuple (lookup)
++ import Data.Foldable (lookup)
+```
+
+Several modules have moved or been renamed, all from the `profunctors` and `bifunctors` packages into the `functors` package. You'll need to update your imports accordingly:
+
+```
+Data.Profunctor.Clown  -> Data.Functor.Clown
+Data.Profunctor.Cowrap -> Data.Functor.Flip
+Data.Profunctor.Joker  -> Data.Functor.Joker
+Data.Profunctor.Costar -> Data.Functor.Costar
+
+Data.Bifunctor.Clown   -> Data.Functor.Clown
+Data.Bifunctor.Flip    -> Data.Functor.Flip
+Data.Bifunctor.Joker   -> Data.Functor.Joker
+Data.Bifunctor.Product -> Data.Functor.Product2
+```
+
+The `Data.Profunctor.Wrap` and `Data.Bifunctor.Wrap` types have been deleted, as any profunctor or bifunctor should also have a functor instance.
+
+### Notable Library Changes
+
+#### The `Newtype` class now has `Coercible` as a superclass and no longer has `wrap` and `unwrap` class members.
+
+Implemented in [purescript-newtype#22](https://github.com/purescript/purescript-newtype/pull/22).
+
+The `Newtype` class exists to support convenient wrapping and unwrapping and to implement the combinators in the `Data.Newtype` module. The `Newtype` class now has `Coercible` as a superclass, which allows most combinators to now be implemented with `coerce` from the new `safe-coerce` library. This makes these combinators zero-cost for a performance gain.
+
+The `Newtype` class no longer has `wrap` and `unwrap` members. Instead, `wrap` and `unwrap` are synonyms for `coerce` and are exported as functions from the module. Most code does not need to change, but if you manually defined a `Newtype` instance in your code it will no longer compile.
+
+In addition, newtypes that do not have their constructors exported can no longer be unwrapped via the `Newtype` class. Most code will be unaffected, as giving a `Newtype` instance to a newtype with hidden constructors was already considered bad practice and usually accidental.
+
+**To fix:**
+
+Most `Newtype` instances are derived and will continue to work. If you manually defined a `Newtype` instance for a type, however, you can simply remove the `wrap` and `unwrap` definitions:
+
+```diff
++ instance newtypeAdditive :: Newtype (Additive a) a
+- instance newtypeAdditive :: Newtype (Additive a) a where
+-   wrap = Additive
+-   unwrap (Additive a) = a
+```
+
+Most newtypes either export their constructors and have a `Newtype` instance, or hide their constructors and do not have the instance. A new compiler warning is raised for types with a `Newtype` instance but hidden constructors.
+
+To address the compiler warning, you can either:
+
+1. Export the newtype constructors, or
+1. Remove the `Newtype` instance -- a breaking change, but the correct option if you used hidden constructors to make your newtype a smart constructor.
+
+#### The `MonadZero` class has been deprecated.
```

Comment:
Compiling core modules exporting MonadZero instances, such as Data.List or Data.Maybe, will yield deprecation warnings until we remove the instances. Should we mention that those warnings can be hidden by installing `psa` and compiling with `psa --censor-codes=UserDefinedWarning`?

Reply:
> Should we mention

Yeah, we should.